.github/workflows/plugin-update.yml

```diff
@@ -0,0 +1,117 @@
+        required: true
+      plugin_tag:
+        type: string
+        description: 'The name of the plugin tag.'
+        required: true
+
+jobs:
+  plugin-update:
+    runs-on: ubuntu-latest
+    env:
+      VAULT_BRANCH: "update/${{ inputs.repo }}/${{ inputs.plugin_tag }}"
+    steps:
+      - run: echo "Branch ${{ inputs.plugin_tag }} of ${{ inputs.repo }}"
+      - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+        with:
+          # We don't use the default token so that checks are executed on the resulting PR
+          # https://docs.github.com/en/actions/using-workflows/triggering-a-workflow#triggering-a-workflow-from-a-workflow
+          token: ${{ secrets.ELEVATED_GITHUB_TOKEN }}
+
+      - uses: actions/setup-go@fac708d6674e30b6ba41289acaab6d4b75aa0753 # v4.0.1
+        with:
+          cache: false # save cache space for vault builds: https://github.com/hashicorp/vault/pull/21764
+          go-version-file: .go-version
+
+      - name: update plugin
+        run: |
+          go get "github.com/${{ inputs.repo }}@${{ inputs.plugin_tag }}"
+          go mod tidy
+
+      - name: detect changes
+        run: |
+          count=$(git status --porcelain=v1 2>/dev/null | wc -l)
+          if [ "$count" -eq 0 ]; then
+            echo "error: no updates were made for repo ${{ inputs.repo }} with tag ${{ inputs.plugin_tag }}"
+            exit 1
+          fi
+
+      - name: commit/push
+        run: |
+          git config user.name hc-github-team-secure-vault-ecosystem
+          git config user.email hc-github-team-secure-vault-ecosystem@users.noreply.github.com
+          git add .
+          git commit -m "Automated dependency upgrades"
+          git push -f origin ${{ github.ref_name }}:"$VAULT_BRANCH"
+
+      - name: Open pull request if needed
+        id: pr
+        env:
+          GITHUB_TOKEN: ${{secrets.ELEVATED_GITHUB_TOKEN}}
+        # Only open a PR if the branch is not attached to an existing one
+        run: |
+          PR=$(gh pr list --head "$VAULT_BRANCH" --json number -q '.[0].number')
+
+          if [ -z "$PR" ]; then
+            gh pr create \
+            --head "$VAULT_BRANCH" \
+            --title "Update ${{ inputs.repo }} to ${{ inputs.plugin_tag }}" \
+            --body "Updates ${{ inputs.repo }} to verify vault CI. Full log: https://github.com/hashicorp/vault/actions/runs/${{ github.run_id }}"
+
+            echo "vault_pr_num=$(gh pr list --head "$VAULT_BRANCH" --json number -q '.[0].number')" >> "$GITHUB_OUTPUT"
+            echo "vault_pr_url=$(gh pr list --head "$VAULT_BRANCH" --json url -q '.[0].url')" >> "$GITHUB_OUTPUT"
+          else
+            echo "Pull request already exists, won't create a new one."
+          fi
+
+      - name: Add changelog
+        continue-on-error: true
+        run: |
+          if [ -z "$vault_pr_num" ]; then
+            echo "error: no vault PR found"
+            exit 1
+          fi
```

Comment:
I _think_ you can use a `ìf:` condition at a step level so, instead of this bash check + continue-on-error, having this condition as a native `ìf:` check would make the execution overview a bit cleaner in the GitHub UI's action panel.

If it's not possible nevermind, what is done here is also a good solution.